server/raft.go

```diff
@@ -1047,7 +1082,37 @@ func (n *raft) decodeAppendEntryResponse(msg []byte) *appendEntryResponse {
 	return ar
 }
 
+// Called when a peer has forwarded a proposal.
+func (n *raft) handleForwardedProposal(sub *subscription, c *client, _, reply string, msg []byte) {
+	if !n.Leader() {
+		n.debug("Ignoring forwarded proposal, not leader")
+		return
+	}
+	// Need to copy since this is underlying client/route buffer.
+	msg = append(msg[:0:0], msg...)
+	if err := n.Propose(msg); err != nil {
+		n.warn("Got error processing forwarded proposal: %v", err)
+	}
+}
+
 func (n *raft) runAsLeader() {
+	n.Lock()
+	// For forwarded proposals.
+	fsub, err := n.subscribe(n.psubj, n.handleForwardedProposal)
+	if err != nil {
+		n.warn("Error subscribing to forwarded proposals: %v", err)
```

Comment:
Can clustering work without this subscription?

Reply:
It can sans forwarded proposals. This should not happen ever, should I change to a panic?

Reply:
Moved err check outside lock and made it a panic for now.